src/plexil-adapter/OwAdapter.cpp

```diff
@@ -52,7 +44,16 @@ static vector<Value> const EmptyArgs;
 
 static int CommandId = 0;
 
-static std::map<int, pair<Command*, AdapterExecInterface*>> CommandRegistry;
+std::mutex g_shared_mutex;
+using CommandRecord = std::tuple<Command*, AdapterExecInterface*, bool>;
```

Comment:
I understand the added boolean flag, but a comment would be helpful in the code.  Also some accessor functions would make use of the tuple easier to understand, ones that wrap the get<n> calls.

Reply:
Agree, I am still not happy with the code yet.